phoenix-core/src/main/java/org/apache/phoenix/compile/CreateTableCompiler.java

```diff
@@ -177,9 +179,22 @@ public MutationPlan compile(CreateTableStatement create) throws SQLException {
                         }
                     }
                 }
+
                 if (isPKMissed) {
-                    throw new SQLExceptionInfo.Builder(SQLExceptionCode.PRIMARY_KEY_MISSING)
-                            .build().buildException();
+                    boolean isTableAvailable = true;
+                    try (HBaseAdmin admin = connection.unwrap(PhoenixConnection.class).getQueryServices().getAdmin()) {
```

Comment:
Instead of an extra admin call to check the base table existence here, wouldn't we be checking this already somewhere in the create view path already? I'm wondering if we can avoid this extra check and just let the exception bubble out from that original place. Maybe we can pass the `isPKMissed` flag so that we know later on whether to throw TNFE or Primary-key-missing SQLException